test/run_test.py

```diff
@@ -71,8 +72,21 @@
     TESTS.extend([
         'rpc_spawn',
         'dist_autograd_spawn',
+        'dist_optimizer_spawn',
     ])
 
+# For tests not in the test/ directory.
+DIR_FOR_TESTS = {
+    "c10d": "distributed",
+    "c10d_spawn": "distributed",
+    "data_parallel": "distributed",
+    "distributed": "distributed",
+    "nccl": "distributed",
+    "rpc_spawn": "distributed/rpc",
+    "dist_autograd_spawn": "distributed/rpc",
+    "dist_optimizer_spawn": "distributed/rpc",
```

Comment:
This is definitely not how you would have written the code to start with ;)

Reply:
Will fix this to find the appropriate directory automatically.